tutorials/nservicebus-sagas/2-timeouts/Solution/Sales/BuyersRemorsePolicy.cs

```diff
@@ -0,0 +1,66 @@
+﻿namespace Core_7.BuyersRemorseTimeoutRequest
+{
+    using NServiceBus;
+    using NServiceBus.Logging;
+    using System;
+    using System.Threading.Tasks;
+    using Messages;
+
+    class BuyersRemorsePolicy
+        : Saga<BuyersRemorseState>
+        , IAmStartedByMessages<PlaceOrder>
+        , IHandleMessages<CancelOrder>
+        , IHandleTimeouts<BuyersRemorseIsOver>
+    {
+        static ILog log = LogManager.GetLogger<BuyersRemorsePolicy>();
+
+        protected override void ConfigureHowToFindSaga(SagaPropertyMapper<BuyersRemorseState> mapper)
+        {
+            mapper.ConfigureMapping<PlaceOrder>(p => p.OrderId).ToSaga(s => s.OrderId);
+            mapper.ConfigureMapping<CancelOrder>(p => p.OrderId).ToSaga(s => s.OrderId);
+        }
+
+        public async Task Handle(PlaceOrder message, IMessageHandlerContext context)
+        {
+            log.Info($"Received PlaceOrder, OrderId = {message.OrderId}");
+
+            log.Info($"Starting cool down period for order #{Data.OrderId}.");
+            await RequestTimeout(context, TimeSpan.FromSeconds(20), new BuyersRemorseIsOver());
+        }
+
+        public async Task Timeout(BuyersRemorseIsOver state, IMessageHandlerContext context)
+        {
+            var orderPlaced = new OrderPlaced
+            {
+                OrderId = Data.OrderId
+            };
+
+            // TODO: Save order state in database?
```

Comment:
I don't get this. The saga is already stored inside the database? What do we need to do? Does `OrderPlaced` not do everything we'd need? People might think we need to actually access our business database and store something?